cmd/prune-junit-xml/prunexml.go

```diff
@@ -135,34 +135,54 @@ func pruneTESTS(suites *junitxml.JUnitTestSuites) {
 		updatedTestcase.Classname = match[1]
 		updatedTestcase.Name = match[2]
 		updatedTestcase.Time = suite.Time
+		updatedSystemOut := ""
+		updatedSystemErr := ""
 		for _, testcase := range suite.TestCases {
 			// The top level testcase element in a JUnit xml file does not have the / character.
 			if testcase.Failure != nil {
 				failflag = true
 				updatedTestcaseFailure.Message = joinTexts(updatedTestcaseFailure.Message, testcase.Failure.Message)
 				updatedTestcaseFailure.Contents = joinTexts(updatedTestcaseFailure.Contents, testcase.Failure.Contents)
 				updatedTestcaseFailure.Type = joinTexts(updatedTestcaseFailure.Type, testcase.Failure.Type)
+				updatedSystemOut = joinTexts(updatedSystemOut, testcase.SystemOut)
+				updatedSystemErr = joinTexts(updatedSystemErr, testcase.SystemErr)
 			}
 		}
 		if failflag {
 			updatedTestcase.Failure = &updatedTestcaseFailure
+			updatedTestcase.SystemOut = updatedSystemOut
+			updatedTestcase.SystemErr = updatedSystemErr
 		}
 		suite.TestCases = append(updatedTestcases, updatedTestcase)
 		updatedTestsuites = append(updatedTestsuites, suite)
 	}
 	suites.Suites = updatedTestsuites
 }
 
-// joinTexts returns "<a>; <b>" if both are non-empty,
+// joinTexts returns "<a><empty line><b>" if both are non-empty,
 // otherwise just the non-empty string, if there is one.
+//
+// If <b> is contained completely in <a>, <a> gets returned because repeating
+// exactly the same string again doesn't add any information. Typically
+// this occurs when joining the failure message because that is the fixed
+// string "Failed" for all tests, regardless of what the test logged.
+// The test log output is typically different because it cointains "=== RUN
+// <test name>" and thus doesn't get dropped.
 func joinTexts(a, b string) string {
 	if a == "" {
 		return b
 	}
 	if b == "" {
 		return a
 	}
-	return a + "; " + b
+	if strings.Contains(a, b) {
+		return a
```

Comment:
The `joinTexts` function is responsible for appending subtest failure details to create a consolidated list.

Each subtest must retain all its details individually. If the function fails to append the details when they are already present, it can result in missing information.

Is there a way to display all the details of a single subtest at once before proceeding to the next? This approach would ensure clarity during the aggregation process.

Please see the below snapshots where the `=== RUN   TestFrontProxyConfig/WithUID` is missing it's failure message.

<img width="788" alt="image" src="https://github.com/user-attachments/assets/1490b6a7-e49b-46fb-80f6-4b5bfc0a44c4">
<img width="847" alt="image" src="https://github.com/user-attachments/assets/2861d858-0f5c-43cc-b2c0-7dd995906f12">

Reply:
> If the function fails to append the details when they are already present, it can result in missing information.

If they are already present, they cannot be missing... Saying the same thing twice without some meta context doesn't add information.

> Please see the below snapshots where the `=== RUN   TestFrontProxyConfig/WithUID` is missing it's failure message.

This is from https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/129035/pull-kubernetes-integration/1863500189038284800, right?

What you want didn't work before either. This used to be rendered as:
```
{Failed; Failed; Failed  === RUN   TestFrontProxyConfig/WithoutUID ... 
...
--- FAIL: TestFrontProxyConfig/WithoutUID (64.89s)
; === RUN   TestFrontProxyConfig/WithUID
```

In other words, the "Failed" summary and the detailed failure messages are detached.

It would be possible to put the test name before the summary and each failure message. I think that would just make the entire output longer without making it more useful. It's already visible what failed.

Reply:
Although this approach looks good to me, there’s a potential risk of mismatches between the count of 'summaries' and 'detailed failure messages.'
This mismatch could lead to ambiguity or difficulty in mapping summaries to their corresponding test failures, especially if the summaries provide crucial information beyond just stating that a test 'failed.'

Please pardon my ignorance if I’m overstating a minor concern.

Reply:
The JUnit file produced by gotestsum always has "Failed" as failure message, and those are the only files that we process with prune-junit-xml.